rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -1407,15 +1410,24 @@ def test_upgrade_db_36_to_37(user_data_dir):  # pylint: disable=unused-argument
         (74, 'state', 1),
     ]
     new_history_events = cursor.execute('SELECT * FROM history_events;').fetchall()
-    for entry in kraken_events:
-        if entry[0] == 176:
-            # this event is modified in two places and is checked later
+    updated_kraken_events = [
+        (171, 0, 'SDASD-DSAD-DSAD', 0, 1638706007439, 'B', 'kraken', 'ETH', '10.996', '11843.777435000000', None, 'withdrawal', 'none'),  # noqa: E501
+        (172, 0, 'YDASD-YDSAD-YDSAD', 0, 1636553589350, 'B', 'kraken', 'ETH', '5.1', '6145.834', None, 'trade', 'spend'),  # noqa: E501
+        (173, 0, 'TXZSDG-IUSNH2-OOAIE3U', 0, 1679243606179, 'B', 'kraken', 'BTC', '0.0922995600', '1391.564299579200', None, 'trade', 'spend'),  # noqa: E501
+        (174, 0, 'TXZSDG-IUSNH2-OOAIE3U', 1, 1679243606179, 'B', 'kraken', 'EUR', '1100', '1391.564299579200', None, 'trade', 'receive'),  # noqa: E501
+        (175, 0, 'TXZSDG-IUSNH2-OOAIE3U', 2, 1679243606179, 'B', 'kraken', 'EUR', '8.00', '8.53312', None, 'trade', 'fee'),  # noqa: E501
+        (176, 0, 'KRAKEN-ETH2-EVENT', 0, 1681826144996, 'B', 'kraken', 'ETH2', '0.0032936117', '6.930681228076', 'Automatic virtual conversion of staked ETH rewards to ETH', 'informational', 'none'),  # noqa: E501
+        (177, 0, 'KRAKEN-ETH-EVENT-STAKING', 0, 1681967948701, 'B', 'kraken', 'ETH', '0.0000355988', '0.069824910272', None, 'staking', 'reward'),  # noqa: E501
+        (1113, 0, 'TSTLG5', 0, 1621508243539, 'B', 'Kraken 1', 'EUR', '37.7000', '0', '', 'trade', 'spend'),  # noqa: E501
+        (1114, 0, 'TSTLG5', 2, 1621508243539, 'B', 'Kraken 1', 'EUR', '0.5500', '0', '', 'spend', 'fee'),  # noqa: E501
+        (1115, 0, 'TSTLG5', 1, 1621508243543, 'B', 'Kraken 1', 'ETH', '0.0169800000', '0', '', 'trade', 'receive'),  # noqa: E501
+    ]
+    for entry in new_history_events:
+        if entry[5] != 'B':
+            # we only check the kraken events here
             continue
-
-        if entry[11] is None and entry[0]:
-            assert (entry[0], 0, *entry[1:11]) + ('none',) in new_history_events
-        else:
-            assert (entry[0], 0, *entry[1:12]) in new_history_events
+        # for kraken events we check also values and subtypes
+        assert entry in updated_kraken_events
```

Comment:
I would compute sets of expected `updated_kraken_events` and the actual entries, and then compare the sets, to make sure that no event was lost

Reply:
Why did you not do this suggestion?
Either do what Alexey suggests or add a counter and check the counter equals length. Otherwise test can miss events.

Reply:
I was waiting to clarify the other issue and alexey pinging for a review before closing everything